prospero-cli/src/main/java/org/wildfly/prospero/cli/RepositoryDefinition.java

```diff
@@ -17,43 +17,93 @@
 
 package org.wildfly.prospero.cli;
 
+import org.jboss.logging.Logger;
 import org.wildfly.channel.Repository;
 
 import java.net.MalformedURLException;
+import java.net.URISyntaxException;
 import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.regex.Pattern;
 
 public class RepositoryDefinition {
 
-    public static List<Repository> from(List<String> repos) throws ArgumentParsingException {
+    public static List<Repository> from(List<String> repos) throws ArgumentParsingException, MalformedURLException, URISyntaxException {
```

Comment:
Can you catch the new exceptions and wrap them in ArgumentParsingException with an explanation message?